modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -1521,6 +1523,40 @@ public CPDefinition fetchCPDefinitionByFriendlyURL(
 			friendlyURLEntry.getClassPK());
 	}
 
+	@Override
+	public List<CPDefinition> findByExpirationDate(
+		Date expirationDate, QueryDefinition<CPDefinition> queryDefinition) {
+
+		return TransformUtil.transform(
+			dslQuery(
+				DSLQueryFactoryUtil.selectDistinct(
```

Comment:
question: why do we always do a distinct? I mean, in this case it's the if of the table and we are not joining with anything so it's "distinct" by default